packages/terra-dialog/src/Dialog.jsx

```diff
@@ -3,39 +3,60 @@ import PropTypes from 'prop-types';
 import classNames from 'classnames/bind';
 import 'terra-base/lib/baseStyles';
 import ContentContainer from 'terra-content-container';
+import ActionHeader from 'terra-clinical-action-header';
+import DialogBody from './DialogBody';
+import DialogFooter from './DialogFooter';
 import styles from './Dialog.scss';
 
 const cx = classNames.bind(styles);
 
 const propTypes = {
- /*
- * The header element to be placed within the header area of the dialog.
- */
-  header: PropTypes.node.isRequired,
-  /*
- * The footer element to be placed within the footer area of the dialog.
- */
-  footer: PropTypes.node,
-  /*
- * The children to be placed within the main content area of the dialog.
- */
+  /**
+   * The text string to be placed within the header area of the dialog.
+   */
+  header: PropTypes.string.isRequired,
+  /**
+   * The footer element to be placed within the footer area of the dialog.
+   * The footer content area will not display if this prop is not provided. UNDER EVALUATION
+   */
+  footer: PropTypes.node.isRequired,
+  /**
+   * The children to be placed within the main content area of the dialog.
+   */
   children: PropTypes.node.isRequired,
```

Comment:
I'd make this optional as well to be honest.

Reply:
What is a dialog with no content? Seems...wrong.

Reply:
It does and I'm sure it's an unlikely case, but it's still an edge case that I'd feel safer having than not. I know for table cells that api is a little confusing in that you can't have an empty table cell.

Reply:
I don't know. That seems like a different situation, in that an empty table cell feels appropriate. That signifies "No data point" rather than "No content".

Reply:
But if a developer passes in null or no children, does it change the display logic of the component? If the dialog can't display if this is null or empty, I'm fine leaving it as required. However, it the dialog handles fine with null or empty children, I still vote to drop the isRequired off this prop. It doesn't change our implementation logic, but still allows for our API to handle that edge case.

Reply:
This conversation is dumb and not important. I have removed the required flag because the impact in either direction is minimal and not worth the effort already invested in the discussion.

https://github.com/cerner/terra-core/pull/1209/commits/74c002d1475e1d182bbab294b424c4d971cae104